### ISSUE TITLE
[BUG] Does ZeRO 1 optimizer affect the convergence of loss or optimization?

### ISSUE THREAD

**Issue by nijkah** [microsoft/DeepSpeed#2365]
## Describe the bug
I'm trying to integrate DeepSpeed into an existing framework. The development is almost done.
Now, I am trying to reproduce the previous experiment with the ZeRO optimizer.
However, I found that the experiment with the ZeRO optimizer cannot reproduce the performance with the same configuration (except using ZeRO optimizer.)

- Does the ZeRO 1 optimizer affect the convergence of loss or optimization?
- Should I tune some hyperparameters when using ZeRO compared to the original experiemnt without ZeRO?

## Additional Information
The experiment I did is training `UPerNet-ResNet50` with `SGD` optimizer. Actually, it doesn't require the ZeRO optimizer to train the model with V100 GPUs. I just try to ensure the integration is well done.

The same experiment with DeepSpeed Runtime without using ZeRO settings showed the same performance compared to the original experiment.

**Comment by tjruwase** [comment]
@nijkah, thanks for reporting this issue. Conceptually, ZeRO should not affect convergence. However, ZeRO has mainly been tested with Adam-like optimizers, but not with SGD. Can you supply steps to reproduce so we can investigate further?  Thanks!

**Comment by nijkah** [comment]
@tjruwase Sorry for late reply.

The reason for poor performance was failure to load pre-trained weights.
I initialized the model and optimizer and passed these to `deepspeed.initialize`.
I load the pre-trained weight after `deepspeed.initialize` and it results in poor performance.

After loading the pre-trained weight before `deepspeed.initialize`, I could see similar convergence and performance with existed experiment.
